File: 프로그래머스/unrated/181894. 2의 영역/2의 영역.java
class Solution {
    public int[] solution(int[] arr) {
        int startindex = -1;
        
        for(int i=0;i<arr.length;i++){
            if(arr[i] == 2){
                startindex = i;
                break;
            }
        }
        
        int endindex = startindex;
        
        for(int i=arr.length-1;i>startindex;i--){
            if(arr[i] == 2){
                endindex = i;
                break;
            }
        }
        
        int[] answer = new int[endindex-startindex+1];
        int index = 0;
        
        
       if(endindex!=-1 &&endindex == startindex) {
    	   answer[0] = 2;
       }else if(endindex ==-1) {
    	   answer[0] = -1;
       }else {
    	   for(int i=0;i<answer.length;i++){
               answer[i] = arr[i+startindex];
           }
       }
              
        return answer;
    }
}

File: 프로그래머스/0/181894. 2의 영역/2의 영역.js
function solution(arr) {
    var answer = [];
    let startIndex = -1;
    let endIndex = -1;
    
    for(let i=0; i<arr.length;i++){
        if(startIndex === -1 && arr[i] === 2){
            startIndex = i;
            continue;
        }
        if(startIndex !== -1  && arr[i] === 2){
            endIndex = i;
            continue;
        }
    }
    
    if(startIndex === -1 && endIndex === -1){
        answer.push(-1);
    }else if(startIndex !== -1 && endIndex === -1){
        answer.push(arr[startIndex]);
    }else {
        for(let i = startIndex; i<=endIndex; i++){
            answer.push(arr[i]);
        }
    }
    
    return answer;
}